### ISSUE TITLE
Wrong autocorrect for `Bundler/OrderedGems` with missing final newline

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#15302]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Bundler/OrderedGems

### Steps to reproduce the problem

```rb
  context 'When the final newline is missing' do
    it 'registers an offense' do
      expect_offense(<<~RUBY, chomp: true)
        gem 'b'
        gem 'a'
        ^^^^^^^ Gems should be sorted in an alphabetical order within their section of the Gemfile. Gem `a` should appear before `b`.
      RUBY

      expect_correction(<<~RUBY.chomp)
        gem 'a'
        gem 'b'
      RUBY
    end
  end
```

### Expected behavior

See above. It just swaps the lines and when the last one doesn't have a newline it causes invalid syntax. Happens with format on save.

### Actual behavior

```
1) RuboCop::Cop::Bundler::OrderedGems When the final newline is missing registers an offense
     Failure/Error: expect(new_source).to eq(correction)
     
       expected: "gem 'a'\ngem 'b'"
            got: "gem 'a'gem 'b'\n"
     
       (compared using ==)
     
       Diff:
       @@ -1,2 +1 @@
       -gem 'a'
       -gem 'b'
       +gem 'a'gem 'b'
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.88.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 4.0.0) +Rubydex [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```
